prow/scripts/cluster-integration/kyma-gke-end-to-end-test.sh

```diff
@@ -0,0 +1,302 @@
+KYMA_RESOURCES_DIR="${KYMA_SOURCES_DIR}/installation/resources"
+
+INSTALLER_YAML="${KYMA_RESOURCES_DIR}/installer.yaml"
+INSTALLER_CONFIG="${KYMA_RESOURCES_DIR}/installer-config-cluster.yaml.tpl"
+INSTALLER_CR="${KYMA_RESOURCES_DIR}/installer-cr-cluster.yaml.tpl"
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+
+shout "Authenticate"
+date
+init
+DNS_DOMAIN="$(gcloud dns managed-zones describe "${CLOUDSDK_DNS_ZONE_NAME}" --format="value(dnsName)")"
+
+if [[ "$BUILD_TYPE" != "release" ]]; then
+    shout "Build Kyma-Installer Docker image"
+    date
+    CLEANUP_DOCKER_IMAGE="true"
+    "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-image.sh"
+fi
+
+shout "Reserve IP Address for Ingressgateway"
+date
+GATEWAY_IP_ADDRESS_NAME="${COMMON_NAME}"
+GATEWAY_IP_ADDRESS=$(IP_ADDRESS_NAME=${GATEWAY_IP_ADDRESS_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/reserve-ip-address.sh")
+CLEANUP_GATEWAY_IP_ADDRESS="true"
+echo "Created IP Address for Ingressgateway: ${GATEWAY_IP_ADDRESS}"
+
+
+shout "Create DNS Record for Ingressgateway IP"
+date
+GATEWAY_DNS_FULL_NAME="*.${DNS_SUBDOMAIN}.${DNS_DOMAIN}"
+CLEANUP_GATEWAY_DNS_RECORD="true"
+IP_ADDRESS=${GATEWAY_IP_ADDRESS} DNS_FULL_NAME=${GATEWAY_DNS_FULL_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-dns-record.sh"
+
+
+shout "Reserve IP Address for Remote Environments"
+date
+REMOTEENVS_IP_ADDRESS_NAME="remoteenvs-${COMMON_NAME}"
+REMOTEENVS_IP_ADDRESS=$(IP_ADDRESS_NAME=${REMOTEENVS_IP_ADDRESS_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/reserve-ip-address.sh")
+CLEANUP_REMOTEENVS_IP_ADDRESS="true"
+echo "Created IP Address for Remote Environments: ${REMOTEENVS_IP_ADDRESS}"
+
+
+shout "Create DNS Record for Remote Environments IP"
+date
+REMOTEENVS_DNS_FULL_NAME="gateway.${DNS_SUBDOMAIN}.${DNS_DOMAIN}"
+CLEANUP_REMOTEENVS_DNS_RECORD="true"
+IP_ADDRESS=${REMOTEENVS_IP_ADDRESS} DNS_FULL_NAME=${REMOTEENVS_DNS_FULL_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-dns-record.sh"
+
+
+shout "Provision cluster: \"${CLUSTER_NAME}\""
+date
+export GCLOUD_SERVICE_KEY_PATH="${GOOGLE_APPLICATION_CREDENTIALS}"
+if [ -z "$MACHINE_TYPE" ]; then
+      export MACHINE_TYPE="${DEFAULT_MACHINE_TYPE}"
+fi
+if [ -z "${CLUSTER_VERSION}" ]; then
+      export CLUSTER_VERSION="${DEFAULT_CLUSTER_VERSION}"
+fi
+CLEANUP_CLUSTER="true"
+"${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/provision-gke-cluster.sh"
+
+shout "Install Tiller"
+date
+kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin --user="$(gcloud config get-value account)"
+"${KYMA_SCRIPTS_DIR}"/install-tiller.sh
+
+
+shout "Generate self-signed certificate"
+date
+DOMAIN="${DNS_SUBDOMAIN}.${DNS_DOMAIN%?}"
+export DOMAIN
+CERT_KEY=$("${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/generate-self-signed-cert.sh")
+TLS_CERT=$(echo "${CERT_KEY}" | head -1)
+TLS_KEY=$(echo "${CERT_KEY}" | tail -1)
+
+shout "Apply Kyma config"
+date
+
+if [[ "$BUILD_TYPE" == "release" ]]; then
```

Comment:
in configuration you specified that BUILD_TYPE is master, so why this condition is here?

Reply:
wiil be removed